### PR TITLE
REACH-148 NodeView for PythonNode

### PR DESCRIPTION
**Pull request by pbidenko**
Implemented NodeView for PythonNode.
Also small fix for List.Create node.

### CODE REVIEW

**Comment by pbidenko** [comment]
Hi @Benglin, please take a look.

File: src/DynamoWebServer/Messages/MessageHandler.cs [hunk 1]
```diff
@@ -522,7 +523,27 @@ private string GetInOutPortsData(NodeModel node)
...
+                    stringBuilder.Append("\", ");
+                }
+
+                stringBuilder.Append("\"varInputs\": [");
+            }
+            else
+            {
+                stringBuilder.Append("\"InPorts\": [");
+            }
+
```
Comment (Benglin) [inline src/DynamoWebServer/Messages/MessageHandler.cs]: Hi @pbidenko, while these changes look good, I am not sure if the details can/should be factored out as extensions? We have multiple extensions defined in *DynamoCore\Utilities\Extensions.cs*, you could add an equivalent for `DynamoWebServer/Extensions.cs`, so the code here can look like:

```csharp
var varNode = node as VariableInputNode;
if (varNode != null)
    stringBuilder.Append(varNode.GetInOutPortsData());
```

Then the same can be applied to `CodeBlockNodeModel` too.

Comment (Benglin) [inline src/DynamoWebServer/Messages/MessageHandler.cs]: Another thing, I noticed the logic around here to be:

```csharp
if (node is CodeBlockNodeModel)
{
    // Does something
}

if (node is VariableInputNode)
{
    // Does something else
}
else
{
    // CodeBlockNodeModel will get here also...
}
```

The `else` will include `CodeBlockNodeModel`, just want to make sure that is expected?

**Comment by Benglin** [comment]
Thanks for the work, @pbidenko. I'm merging this in if you could clarify the above `else` statement. 

Apart from that, I'm wondering if we could refactor `MessageHandler` class in a separate PR? I'm thinking the following can be done:

1. `uploader` may still be made `readonly`.
2. The class layout can be updated:

   - Data members
   - Properties
   - Public, internal, protected, private methods

3. `MESSAGE_TYPE` should be renamed as `MessageTypeFormat` and move into a separate `Constants.cs` file (we could move more `DynamoWebServer` strings there in near future). Here it will be referred to as `Constants.MessageTypeFormat`.

4. Few string constants can be moved to `Constants.cs` too, e.g. `MyCustomNode`, `Home`, `.dyf`, etc.

5. `Serialize` could be renamed as `SerializeMessage` to align better with `DeserializeMessage`.

Hope that makes sense?

**Comment by pbidenko** [comment]
Hi @Benglin,
About 'else' statement - it is expected behavior.
And for the rest, refactor MessageHandler class sounds reasonable, thank you for reviewing. I've opened a new task http://adsk-oss.myjetbrains.com/youtrack/issue/REACH-180

File: src/DynamoWebServer/Messages/MessageHandler.cs (hunk 1 above)
Comment (pbidenko) [inline src/DynamoWebServer/Messages/MessageHandler.cs]: Added DynamoWebServer/Extensions.cs with extensions for CodeBlockNodeModel, VariableInputNode and NodeModel.

**Comment by Benglin** [comment]
Thanks a lot for accommodating, @pbidenko. LGTM!

**Pull request merged**